client/transaction-pool/src/lib.rs

```diff
@@ -440,6 +440,7 @@ async fn prune_known_txs_for_block<Block: BlockT, Api: ChainApi<Block = Block>>(
 	block_id: BlockId<Block>,
 	api: &Api,
 	pool: &sc_transaction_graph::Pool<Api>,
+	pruned_log: &mut HashSet<ExtrinsicHash<Api>>,
```

Comment:
You remove this and return a `Vec<ExtrinsicHash<Api>>` as function result.

Reply:
We should at least try to avoid allocations?

Reply:
We already allocate this vector in the function and just use the slice. So, we don't do any double allocations.

Reply:
Right 🤦‍♀️